In [ ]:
import pandas as pd

In [2]:
df = pd.read_csv('hr.csv')

# número exacto de empleados en cada situación

In [ ]:
df['Attrition'].value_counts()

Attrition
No     1236
Yes     238
Name: count, dtype: int64

# lo pasamos a formato porcentaje

In [11]:
print (df['Attrition'].value_counts(normalize=True) * 100)

print ("Esto significa que ABC Corporation tiene una fuga de talento real. Aunque no es una crisis absoluta, también hay un desbalanceo de clases (un 83.8% de 'No' frente a un 16.1% de 'Yes'), algo importante para un modelo predictivo.")

Attrition
No     83.85346
Yes    16.14654
Name: proportion, dtype: float64
Esto significa que ABC Corporation tiene una fuga de talento real. Aunque no es una crisis absoluta, también hay un desbalanceo de clases (un 83.8% de 'No' frente a un 16.1% de 'Yes'), algo importante para un modelo predictivo.


# Revisión de columnas clave

In [12]:
# Definimos la lista de columnas que toca analizar (Pareja B)
columnas_categoricas = ['JobRole', 'Department', 'Gender', 'MaritalStatus']

# Iteramos por cada una para imprimir sus valores únicos sin repetir
for col in columnas_categoricas:
    print(f"--- Valores únicos en la columna: {col} ---")
    print(df[col].unique())
    print("\n") # Esto es solo para dejar un espacio en blanco entre columnas



--- Valores únicos en la columna: JobRole ---
<StringArray>
[          ' sALES eXECUTIVE ',        ' rESEARCH sCIENTIST ',
     ' lABORATORY tECHNICIAN ',    ' mANUFACTURING dIRECTOR ',
 ' hEALTHCARE rEPRESENTATIVE ',                   ' mANAGER ',
      ' sALES rEPRESENTATIVE ',         ' rESEARCH dIRECTOR ',
           ' hUMAN rESOURCES ']
Length: 9, dtype: str


--- Valores únicos en la columna: Department ---
<StringArray>
['Sales', 'Research & Development', nan, 'Human Resources']
Length: 4, dtype: str


--- Valores únicos en la columna: Gender ---
<StringArray>
['Female', 'Male']
Length: 2, dtype: str


--- Valores únicos en la columna: MaritalStatus ---
<StringArray>
['Single', 'Married', 'Divorced', nan, 'Marreid']
Length: 5, dtype: str




En columnas clave vemos que en JobRole están mal nombradas las columnas, usando la primera con minuscula y las siguientes en mayúsculas. 
En Departament tenemos un nulo, lo que significa que hay trabajadores que no sabemos en que departamento trabajan.
En MaritalStatus tenemos un nulo, por lo que no sabemos su estado civil y además tenemos Married/Marreid lo que origina dos grupos para una misma cosa, el de Marreid casi siempre quedará sin contar (sueles buscar Married)

# contar valores unicos para quitar columnas que no aporten valor (si todos los empleados tienen que ser mayores
# de edad, no tiene sentido una columna de "mayor de edad" en la que siempre ponga si)

In [6]:
df.nunique().sort_values()

EmployeeCount                  1
Over18                         1
StandardHours                  1
Attrition                      2
OverTime                       2
PerformanceRating              2
Gender                         2
BusinessTravel                 3
Department                     3
JobSatisfaction                4
RelationshipSatisfaction       4
StockOptionLevel               4
MaritalStatus                  4
EnvironmentSatisfaction        4
JobInvolvement                 4
WorkLifeBalance                4
Education                      5
JobLevel                       5
EducationField                 6
TrainingTimesLastYear          7
JobRole                        9
NumCompaniesWorked            10
PercentSalaryHike             15
YearsSinceLastPromotion       16
YearsWithCurrManager          18
YearsInCurrentRole            19
DistanceFromHome              29
YearsAtCompany                37
TotalWorkingYears             40
Age                           43
HourlyRate

Eliminaría las columnas EmployeeCount, Over18, StandardHours, para las que su valor siempre es 1 y no aportan información extra valiosa.

### 📋 Reporte de Hallazgos - Fase 1 (Pareja B)

1. **Fuga de Talento (Variable Objetivo):**
   - El **16.15%** de los empleados de la muestra han abandonado la empresa (`Attrition = Yes`). Se confirma la necesidad del análisis visual para detectar los motivos de la fuga.

2. **Problemas detectados en columnas categóricas (Para limpiar en Fase 2):**
   - **`JobRole`:** Formato de texto incorrecto (ej. `sALES eXECUTIVE`) y presencia de espacios en blanco al inicio y final de las cadenas.
   - **`Department`:** Presencia de valores nulos (`nan`).
   - **`MaritalStatus`:** Presencia de valores nulos (`nan`) y una errata de escritura (`'Marreid'`).

3. **Columnas Constantes a Eliminar:**
   - `EmployeeCount`, `Over18` y `StandardHours` tienen un único valor único (`1`). No aportan información al análisis.